src/fetch.rs

```diff
@@ -29,117 +32,118 @@ pub fn fetch(args: InputArgs, filter_list: Vec<Regex>) {
     let filter_hidden = matches.is_present("traverse-hidden");
 
     let dir_tree_with_options = DirectoryTreeOptions {
-        filter_list: filter_list,
-        filter_hidden: filter_hidden,
+        filter_list,
+        filter_hidden,
     };
 
     let root_path = args.get_root_path("PATH");
-    let root = root_path.to_str().expect(format!("{}", "Error in converting directory to string".red()).as_str());
-
-    dir_tree_with_options.process_directories(root, process_directory).unwrap_or_else(|err| {
-        println!("{} {}: {}", "Failed fetching for".red(), root.red(), err);
-        process::exit(1);
-    });
-}
-
-fn process_directory(dir: &DirEntry) -> Result<(), Box<dyn Error>> {
-    if dir.file_name().eq(".git") {
-        match dir.path().parent() {
-            Some(dir) => {
-                let repo = Repository::open(dir)?;
-                fetch_repo(repo)?;
-            }
-            None => {
-                println!("{} {:#?}", "error accessing parent directory of".red(), dir.path())
-            }
-        }
-    }
-    Ok(())
-}
-
-fn fetch_repo<'a>(repo: Repository) -> Result<(), Box<dyn Error>> {
-    let path = repo.path().parent().expect("Failed to get path from repo");
-    let remotes = repo.remotes()?;
-
-    // TODO: handle all remotes
-    if remotes.iter().any(|remote| remote == Some("origin")) {
-        let remote = "origin";
-        print!("{} {} {} {:#?} -> ", "\nFetching".blue(), remote.blue(), "for".blue(), path);
```

Comment:
I moved the remote name up to the fetch function, so we can change it for every repo. I'm not sure of what will happen if we try to pull different origin from the same repo at the same time ? But as long as they are only one remote per repo, it should be safe.

Reply:
We wont pull at same time, it will fetch for one remote only but we will provide an option to fetch for a diffrent remote other than origin. At that time we need to show whihc remote we are pulling for.

Reply:
It now print the remote before the file path, and it's controlled in the parameter of the GitAction, so it's easy to add a Clap flag that control it